packages/button/src/utils.tsx

```diff
@@ -0,0 +1,90 @@
+import { outline, themeValues, tokens, mapSpacing } from '@ag.ds-next/core';
+
+const variants = {
+	primary: {
+		background: themeValues.foreground.action,
+		border: `3px solid ${themeValues.foreground.action}`,
+		color: themeValues.background.page,
+		textDecoration: 'none',
+
+		'&:hover': {
+			background: themeValues.foreground.text,
+			border: `3px solid ${themeValues.foreground.text}`,
+			color: themeValues.background.page,
+			textDecoration: 'underline',
+		},
+
+		'&:focus': { ...outline },
+	},
+	secondary: {
+		background: 'transparent',
+		border: `3px solid ${themeValues.foreground.action}`,
+		color: themeValues.foreground.action,
+		textDecoration: 'none',
+
+		'&:hover': {
+			background: 'transparent',
+			border: `3px solid ${themeValues.foreground.text}`,
+			color: themeValues.foreground.text,
+			textDecoration: 'underline',
+		},
+
+		'&:focus': { ...outline },
+	},
+	tertiary: {
+		background: 'transparent',
+		border: `3px solid transparent`,
+		color: themeValues.foreground.action,
+		textDecoration: 'underline',
+
+		'&:hover': {
+			background: 'transparent',
+			border: `3px solid transparent`,
+			color: themeValues.foreground.text,
+			textDecoration: 'none',
+		},
+
+		'&:focus': { ...outline },
+	},
+};
+
+export type ButtonVariant = 'primary' | 'secondary' | 'tertiary';
+
+export function buttonStyles({
+	block,
+	disabled,
+	variant = 'primary',
+}: {
+	block?: boolean;
+	disabled?: boolean;
+	variant?: ButtonVariant;
+}) {
+	return {
+		...variants[variant],
+		display: 'inline-block',
```

Comment:
```suggestion
		display: 'inline-block',
		borderWidth: 3,
		borderStyle: 'solid',
```

Reply:
We can define the width and style of the border here, which means we only need to change the colors in the `variants` object